app/src/main/java/com/example/riden/activities/RideDetailActivity.java

```diff
@@ -0,0 +1,122 @@
+package com.example.riden.activities;
+
+import android.os.Bundle;
+import android.widget.ImageButton;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.bumptech.glide.Glide;
+import com.example.riden.R;
+import com.example.riden.activities.directionhelpers.TaskLoadedCallback;
+import com.example.riden.models.Ride;
+import com.example.riden.models.User;
+import com.google.android.gms.maps.CameraUpdate;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.MapFragment;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.maps.model.Polyline;
+import com.example.riden.activities.directionhelpers.FetchURL;
+import com.google.android.gms.maps.model.PolylineOptions;
+
+import java.sql.Driver;
+import java.util.ArrayList;
+//import com.thecodecity;
+
+public class RideDetailActivity extends AppCompatActivity implements OnMapReadyCallback, TaskLoadedCallback {
+
+    GoogleMap map;
+    MarkerOptions pickupLocation, destinationLocation;
+    Polyline currentPolyline;
+    TextView tvPickupAddress;
+    TextView tvDestinationAddress;
+    TextView tvDepartureDate;
+    TextView tvDepartureTime;
+    ImageButton ibDriverProfileDetail;
+    ImageButton ibDriverCarDetail;
+
+    Ride ride;
+    User driver;
+
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_ride_detail);
+        MapFragment mapFragment = (MapFragment) getFragmentManager().findFragmentById(R.id.mapFrag);
+        mapFragment.getMapAsync(this);
+
+        tvPickupAddress = findViewById(R.id.tvPickupAddress);
+        tvDestinationAddress = findViewById(R.id.tvDestinationAddress);
+        tvDepartureDate = findViewById(R.id.tvDepartureDate);
+        tvDepartureTime = findViewById(R.id.tvDepartureTime);
+        ibDriverProfileDetail = findViewById(R.id.ibDriverProfileDetail);
+        ibDriverCarDetail = findViewById(R.id.ibDriverCarDetail);
+
+        ride = getIntent().getParcelableExtra(Ride.class.getSimpleName());
+        driver = (User) ride.getDriver();
+
+        pickupLocation = new MarkerOptions().position(new LatLng(ride.getPickupLat(), ride.getPickupLong())).title("Pickup Location");
```

Comment:
Why doesn't ride store pickup Latitude longitude as a LatLng object? Wouldn't that make this simpler.